### DISCUSSION TITLE
服务端和客户端身份校验

### DISCUSSION THREAD

**Discussion by xing403** [nuxt/nuxt#27440] [Questions]
最近使用 Nuxt 打算做一个 blog，
在用户请求登录后，服务端将返回一个`token`，我将这个token存储在了 `localStorage` 使用户在下次访问不用再次登录了.
但是在使用`localStorge.setItem('token', '...')`时，报错`localStorge` 未定义
![image](https://github.com/nuxt/nuxt/assets/74746021/ebab85bb-4147-40ff-8f7d-cc4dba289d50)

替换使用 vueuse 的 useLocalStorage('token', '') 可以正常把token存储在 localStorage, 但是当我刷新浏览器后存储的 `token` 却被自动替换 成了 `''`，
是否有别的方案可以进行身份校验，或者阻止 token 值被覆盖。

**Comment by undermoonn** [comment]
建议用 `useCookie` 存到 document.cookie 中，这是最普遍的 Web 处理登录态方式。此外 localStorage 是 browser API，Node.js 是没有的。

> https://developer.mozilla.org/zh-CN/docs/Web/HTTP/Cookies

**Comment by xing403** [comment]
好的非常感谢！！

**Comment by FatNerdPeng** [comment]
你需要理解，nuxt默认是SSR模式 ，也就是服务端渲染，服务端环境是没有localStorage(这个东西只在浏览器环境中有)。也就是说，你只需要在非SSR期间，就能使用。在setup语法糖中，setup钩子是服务端渲染，也就是说，如果你这么写：
```
<script setup>
localStorage.setItem()
</script>
```
这样就会报错`localStorage`未定义，但是你这么写：
```
<script setup>
function login() {
  localStorage.setItem()
}
</script>

<template>
<div>
<form @submit="login">
</form>
</div>
</template>
```
你在触发表单时，就能使用`localStorage`了。

**Comment by bbhxwl** [comment]
国内的人用的比较少呢。
